packages/core/src/typings/overrides.ts

```diff
@@ -0,0 +1,267 @@
+import type { PropsWithChildren } from 'react'
+import type {
+  AlertProps,
+  BannerTextContentProps,
+  BannerTextProps,
+  BreadcrumbProps,
+  ButtonProps,
+  DiscountBadgeProps,
+  EmptyStateProps,
+  HeroHeaderProps,
+  HeroImageProps,
+  HeroProps,
+  IconButtonProps,
+  IconProps,
+  ImageGalleryProps,
+  ImageZoomProps,
+  InputFieldProps,
+  LinkButtonProps,
+  NavbarButtonsProps,
+  NavbarHeaderProps,
+  NavbarLinksListProps,
+  NavbarLinksProps,
+  NavbarProps,
+  NavbarRowProps,
+  NavbarSliderContentProps,
+  NavbarSliderFooterProps,
+  NavbarSliderHeaderProps,
+  NavbarSliderProps,
+  PriceProps,
+  ProductShelfProps,
+  ProductTitleProps,
+  QuantitySelectorProps,
+  RegionBarProps,
+  ShippingSimulationProps,
+  SkeletonProps,
+  SkuSelectorProps,
+} from '@faststore/ui'
+
+import type {
+  ComponentOverrideDefinition,
+  SectionOverrideDefinition,
+} from './overrideDefinitionUtils'
+
+export type SectionOverride =
+  | AlertOverrideDefinition
+  | BannerTextOverrideDefinition
+  | BreadcrumbOverrideDefinition
+  | EmptyStateOverrideDefinition
+  | HeroOverrideDefinition
+  | NavbarOverrideDefinition
+  | NewsletterOverrideDefinition
+  | ProductDetailsOverrideDefinition
+  | ProductGalleryOverrideDefinition
+  | ProductShelfOverrideDefinition
+  | RegionBarOverrideDefinition
+
+/**
+ * Originally, these types were defined in their respective Overrides file
+ * For some reason, typescript wouldn't interpret SectionOverride correctly when
+ * defining it by importing each of the SectionOverrideDefinitions and would set it as 'any'
+ *
+ * For some reason, defining them in the same file as SectionOverride seems to fix the issue.
+ * Consider that before moving them elsewhere and test it on the starter as well.
+ */
+export type AlertOverrideDefinition = SectionOverrideDefinition<
+  'Alert',
+  {
+    Alert: ComponentOverrideDefinition<AlertProps, Omit<AlertProps, 'onClose'>>
+    Icon: ComponentOverrideDefinition<IconProps, IconProps>
+  }
+>
+
+export type BannerTextOverrideDefinition = SectionOverrideDefinition<
+  'BannerText',
+  {
+    BannerText: ComponentOverrideDefinition<BannerTextProps, BannerTextProps>
+    BannerTextContent: ComponentOverrideDefinition<
+      BannerTextContentProps,
+      BannerTextContentProps
+    >
+  }
+>
+
+export type BreadcrumbOverrideDefinition = SectionOverrideDefinition<
+  'Breadcrumb',
+  {
+    Breadcrumb: ComponentOverrideDefinition<BreadcrumbProps, BreadcrumbProps>
+    Icon: ComponentOverrideDefinition<IconProps, IconProps>
+  }
+>
+
+export type EmptyStateOverrideDefinition = SectionOverrideDefinition<
```

Comment:
We still need to integrate this section with our CMS. Is it ok to keep it even when this is not being used? Or can we remove it for now? I'm planning to remove this section from our [documentation](https://faststore-site-git-docs-updates-override-docs-1-faststore.vercel.app/docs/reference-guides/list-of-native-sections).

Reply:
I think so, yes! At least people can customize it by overriding it.